site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -406,11 +406,18 @@ public function ajaxUploadCourseMaterialsFiles() {
                         $zip = new \ZipArchive();
                         $res = $zip->open($uploaded_files[1]["tmp_name"][$j]);
                         if ($res === true) {
+                            //don't update all the files that were previouslly uploaded
+                            $subfiles_before = FileUtils::getAllFiles($upload_path, array(), true );
                             $zip->extractTo($upload_path);
                             $zip->close();
                             $subfiles = FileUtils::getAllFiles($upload_path, array(), true );
                             foreach ($subfiles as $file) {
-                                $json[$file['path']] = array('checked' => '1', 'release_datetime' => $release_time  );
+                                //skip the files that aren't new
+                                if ( in_array($file, $subfiles_before))
+                                    continue;
+
+                                $json[$file['path']] = array('checked' => '1',
+                                'release_datetime' => $release_time  );
```

Comment:
If not going to have this all on one line, then have opening/closing braces line up with each array entry on a newline:
```
                                $json[$file['path']] = [
                                    'checked' => '1',
                                    'release_datetime' => $release_time
                                ];
```

Reply:
Ended up going through the zip file before it was extracted and getting all the filenames from that, it was too messy extracting to a tmp file and then correcting the paths to the real upload_path